packages/wallet-adapter-react/src/WalletProvider.tsx

```diff
@@ -186,18 +186,21 @@ export const AptosWalletAdapterProvider: FC<AptosWalletProviderProps> = ({
     }
   };
 
-  const changeNetwork = async (network: Network) => {
+  const changeNetwork = async (network: Network, chainId?: number) => {
     if (!walletCore) {
       throw new Error("WalletCore is not initialized");
     }
+    
     try {
-      return await walletCore?.changeNetwork(network);
+      return chainId !== undefined 
```

Comment:
we dont really need this conditional logic as we accept an undefined customChainId and handle it in the wallet-core logic https://github.com/aptos-labs/aptos-wallet-adapter/pull/425/files#diff-8a76cc616d7961765eb7ff494b4e86bc0e56cce6d0273caaa57af714ac95757eR1178
```suggestion
      return await walletCore.changeNetwork(network, chainId)
```